pyVmomi/SoapAdapter.py

```diff
@@ -1237,20 +1261,13 @@ def __init__(self, host='localhost', port=443, ns=None, path='/sdk',
       else:
          self.thumbprint = None
 
-      self.is_ssl_tunnel = False
+      self.is_tunnel = False
       if sslProxyPath:
          self.scheme = SSLTunnelConnection(sslProxyPath)
-         self.is_ssl_tunnel = True
+         self.is_tunnel = True
       elif httpProxyHost:
-         if self.scheme == _HTTPSConnection:
-            self.scheme = SSLTunnelConnection(self.host)
```

Comment:
Hi @goneri , Please can you explain how HttpProxyConnection() is diffrent here from SSLTunnelConnection? SSLTunnelConnection() also works with httpProxyHost. 
Please can you try using SSLTunnelConnection() with httpProxyHost?

Thanks

Reply:
- `HttpProxyConnection()` relies on a HTTP proxy like Squid or TinyProxy. It has to accept HTTP connection and the CONNECT verb. In the context of HTTPS, the proxy will just forward the connection to the final host.

    See: https://en.wikipedia.org/wiki/HTTP_tunnel#HTTP_CONNECT_method

- `SSLTunnelConnection` is different, the tunnel is established between the client and a HTTPS server. The client has to send traffic to a given end-point of the HTTPS, the traffic will end up being forwarded to the final application. So I believe it's similar to haproxy SSL passthrough for instance.